wordpress-mobile/WordPress-Android
# Automate updating gutenberg translations (Take 2)

Fixes wordpress-mobile/gutenberg-mobile#939

Gutenberg PR: https://github.com/wordpress-mobile/gutenberg-mobile/pull/1589

This PR automates adding gutenberg mobile strings to the main `strings.xml` files in order to have those translated by glotpress. It also adds support for plurals in `GutenbergEditorFragment`

### Take 2
**A couple things went wrong in [the previous PR](https://github.com/wordpress-mobile/WordPress-Android/pull/10719) and was detected after it got merged to the gutenberg release branch:**
**- There was a bug to the code that generates plurals strings that caused a crash in production, this is fixed with 996d380**
**- This approach generates new names for gutenberg strings, thus we need to have the main (english) `strings.xml` uploaded to glotpress first in order to have the strings updated for the other languages.**

The latter is a bit problematic as I believe it means we'll lose the current translations we have for gutenberg strings and will have to wait for the new ones to be translated.

### Testing Instructions
- Check the main strings.xml at `WordPress/src/main/res/values/strings.xml` and make sure it contains all the strings from `gutenberg-mobile/bundle/android`
- Build and run the app (the release version preferably) to make sure the main xml is valid (`bundle exec fastlane build_beta skip_confirm:true` and `adb install build/wpandroid-13.6-rc-2-universal.apk`)

PR submission checklist:

- [ ] Update the [release process](https://wp.me/P9ugOq-vy) when the PR lands.
